### DISCUSSION TITLE
Why not run playwright tests in isolation?

### DISCUSSION THREAD

**Discussion by zwhitchcox** [epicweb-dev/epic-stack#767] [Q&A]
I did not see any reason for this in the decisions folder, and so I looked in the [fullstack testing repo](https://github.com/epicweb-dev/full-stack-testing/tree/main/exercises/11.test-db#isolation), and I found

>In our end-to-end tests, we decided to run our tests on the same database as our development server. This comes with a number of nice benefits, but it also means we need to make sure we delete any data we create to avoid polluting our development database with tons of test data.

But I don't understand the benefits and can't find them listed anywhere.

The only reason I can think of would be that you only have to run one testing server, but just wanted to verify if that's it?

We're running into issues with not having the tests run in isolation, and some test suites actually depend on not running in isolation. But why not have a single test db per test file like in `vitest`?

Admittedly I'm new to playwright and have never put much emphasis on e2e tests, so there might be a simple solution to this, so pardon my ignorance

**Comment by kentcdodds** [comment]
I could be convinced that having a separate database for the end-to-end test could be valuable. However you would not be able to do an individual test for each end to end test with playwright. You wouldn't want to spin up a separate instance of the app for every one of those tests. Additionally there's a lot of value in having all of the tests running at the same time against the same instance of the app and database to ensure that your application can handle that kind of load. 

So if someone wants to put together an example of running playwrights against a database that is separate from the development database, I would be open to looking into that and seeing whether it's reliable.

**Comment by zwhitchcox** [comment]
Ok, closing for now, until we get our MVP out
